fix: Don't attempt to include super-long synth logs in a PR body

dazuma opened the pull request:
We recently had an Elixir synth fail (https://github.com/googleapis/elixir-google-api/issues/5748) because the synth log was long enough to trigger GitHub's PR body length limit of 2^16 characters. This PR just checks whether the synth log is getting long-ish, and if so it falls back to the behavior of just including a link to the log rather than including the entire log inline.

dazuma commented:
I don't really feel strongly either way. I don't think I've ever cared about the synth log unless something went wrong.

chingor13 commented:
> I don't think I've ever cared about the synth log unless something went wrong.

IIRC @jskeet cared because the .NET synthesis runs a API change detector and logs the output. Maybe the link to the log is sufficient?

dazuma commented:
I guess if people do care, then it's more convenient to have at least some of it inline. I'll change it to include the last n characters.

dazuma commented:
_and_ include the link

jskeet commented:
I'm confused about the issue - synth logs aren't *in* the PR at the moment, are they? Here's a recent one for C#:

https://github.com/googleapis/google-cloud-dotnet/pull/5220

No log there... what am I missing?

dazuma commented:
@jskeet It looks like it depends. For some language synth scripts, it definitely includes the logs inline (example from Elixir: https://github.com/googleapis/elixir-google-api/pull/5867)

jskeet commented:
Gosh, that would be *so* useful. Okay, will need to work out what triggers that difference at some point.

dazuma commented:
@chingor13 Okay, updated. The behavior is now:

* If the synth log is longer than 40k characters, the beginning is truncated, and the note `[LOG TRUNCATED]` is prepended.
* The link to the log location is included, if available, even if the log (truncated or not) is included inline.
* Updated the tests accordingly, and added a test for the truncation case.

PTAL.
